nativescript-angular/router/page-router-outlet.ts

```diff
@@ -242,7 +248,10 @@ export class PageRouterOutlet { // tslint:disable-line:directive-class-suffix
         // Add it to the new page
         page.content = componentView;
 
-        page.on("navigatedFrom", (<any>global).Zone.current.wrap((args: NavigatedData) => {
+        page.on(Page.navigatedToEvent, () => setTimeout(() => {
```

Comment:
Technically we don't BUT what I want to achieve is to push the actually execution to the end of the event loop so when the view tree to destroy is huge the main thread has time to actually do other things before the clean up happens.